src/SDK/Trace/SpanProcessor/BatchSpanProcessor.php

```diff
@@ -31,27 +32,41 @@ class BatchSpanProcessor implements SpanProcessorInterface
 
     public function __construct(
         ?SpanExporterInterface $exporter,
-        API\ClockInterface $clock,
-        int $maxQueueSize = 2048,
-        int $scheduledDelayMillis = 5000,
-        // @todo: Please, check if this code is needed. It creates an error in phpstan, since it's not used
-        int $exporterTimeoutMillis = 30000,
-        int $maxExportBatchSize = 512
+        API\ClockInterface $clock = null,
+        int $maxQueueSize = null,
+        int $scheduledDelayMillis = null,
+        int $exporterTimeoutMillis = null,
+        int $maxExportBatchSize = null
     ) {
-        if ($maxExportBatchSize > $maxQueueSize) {
-            throw new InvalidArgumentException("maxExportBatchSize should be smaller or equal to $maxQueueSize");
+        if (null === $clock) {
+            $clock = \OpenTelemetry\SDK\Trace\AbstractClock::getDefault();
         }
         $this->exporter = $exporter;
         $this->clock = $clock;
-        $this->maxQueueSize = $maxQueueSize;
-        $this->scheduledDelayMillis = $scheduledDelayMillis;
-        // @todo: Please, check if this code is needed. It creates an error in phpstan, since it's not used
-        $this->exporterTimeoutMillis = $exporterTimeoutMillis;
-        $this->maxExportBatchSize = $maxExportBatchSize;
+        $this->maxQueueSize = $maxQueueSize ?: $this->fromEnv('OTEL_BSP_MAX_QUEUE_SIZE', 2048);
+        $this->scheduledDelayMillis = $scheduledDelayMillis ?: $this->fromEnv('OTEL_BSP_SCHEDULE_DELAY', 5000);
+        $this->exporterTimeoutMillis = $exporterTimeoutMillis ?: $this->fromEnv('OTEL_BSP_EXPORT_TIMEOUT', 30000);
+        $this->maxExportBatchSize = $maxExportBatchSize ?: $this->fromEnv('OTEL_BSP_MAX_EXPORT_BATCH_SIZE', 512);
+        if ($this->maxExportBatchSize > $this->maxQueueSize) {
+            throw new InvalidArgumentException("maxExportBatchSize should be smaller or equal to $this->maxQueueSize");
+        }
 
         $this->queue = [];
     }
 
+    private function fromEnv(string $key, int $default): int
+    {
+        $value = getenv($key);
+        if (false === $value) {
+            return $default;
+        }
+        if (!is_numeric($value)) {
+            throw new Exception($key . ' is not numeric');
+        }
+
+        return (int) $value;
```

Comment:
This casting to int is redundant, since it's throwing a exeption before, when it's not.

Reply:
It's actually required, because getenv always returns a string, and if you don't cast it you get a TypeError. You could argue that the exception is not required, but I thought the meaningful exception message was better than a type error.